content/blog/2019/knative-activator-adapter/index.md

```diff
@@ -0,0 +1,93 @@
+publishdate: 2019-07-18
+attribution: Idan Zach (IBM)
+keywords: [mixer,adapter,knative,scale-from-zero]
+---
+
+This post demonstrates how you can leverage [Mixer](https://istio.io/faq/mixer/) to push application logic
+into Istio. It describes a Mixer adapter which implements the [Knative](https://knative.dev/) scale-from-zero logic
+with a very simple code and similar performance to the original implementation.
+
+## Knative serving
+
+[Knative Serving](https://knative.dev/docs/serving/) builds on [Kubernetes](https://kubernetes.io/) to support deploying
+and serving of serverless applications. A core capability of serverless platforms is scale-to-zero
+functionality which reduces resource usage and cost of inactive workloads. However, this requires a new mechanism - when
+an idle application receives a new request and needs to scale back from zero. Knative scale-to/from-zero is built on
+a number of internal components with a precise and complex interaction.
+
+The diagram below presents the current Knative architecture for scale-from-zero.
+
+{{< image width="60%" link="knative-activator.png" caption="Knative scale-from-zero" >}}
+
+The traffic for an idle application is redirected to **Activator** component by programming Istio with *VirtualServices*
+and *DestinationRules*. When **Activator** receives a new request, it:
+
+1. buffers incoming requests
+
+1. triggers to **Autoscaler**
+
+1. redirects requests to the application after it has been scaled up, including retries and load-balancing (if needed)
+
+Once the application is up and running again, Knative restores the routing from **Activator** to the running application.
+
+## Mixer adapter
+
+[Mixer](https://istio.io/faq/mixer/) provides a rich intermediation layer between the Istio components and infrastructure backends.
+It is designed as a stand-alone component, separate from [Envoy](https://www.envoyproxy.io/), and has a simple extensibility model
+to allow Istio to interoperate with as widest breath of backends as possible. Due to its design, Mixer is inherently easier to extend
+than Envoy is.
+
+It is an attribute processing engine that uses operator-supplied configuration to map request attributes from the proxy into calls
+to these backend systems via a pluggable set of adapters. Adapters enable **Mixer** to expose a single consistent API, independent of the
+infrastructure backends in use. The exact set of adapters used at runtime is determined through operator configuration and can easily
+be extended to target new or custom infrastructure backends.
+
+Here we utilized these benefits to implement the Knative scale-from-zero logic in a special Mixer
+[out-of-process adapter](https://github.com/istio/istio/wiki/Mixer-Out-Of-Process-Adapter-Dev-Guide)
+which allows you to use any programming language and to build and maintain your extension as a stand-alone
+program without the need (as in previous Istio releases) to build the whole Istio proxy.
```

Comment:
I'd remove tghe (as in previous Isito releases) and add a concluding sentence or preceding sentence stating "Prior Istio releases required building the istio proxy.  (Is this accurate?)

Also "build the whole Istio proxy" -> "Build the Istio proxy".